src/runtime/dsl.ts

```diff
@@ -0,0 +1,431 @@
+//--------------------------------------------------------------------
+// Javascript DSL for writing Eve programs
+//--------------------------------------------------------------------
+
+// There don't seem to be TypeScript definitions for these by default,
+// so here we are.
+declare var Proxy:new (obj:any, proxy:any) => any;
+declare var Symbol:any;
+
+import {RawValue, Register, isRegister, GlobalInterner, Scan, IGNORE_REG, ID,
+        InsertNode, Node, Constraint, FunctionConstraint} from "./runtime";
+import * as runtime from "./runtime";
```

Comment:
I started out directly importing everything and then realized I was going to only use several of these once, so it was a bit of a waste. Probably worth cleaning up some more of these that are only used once just for clarity.